### ISSUE TITLE
Add option to not select input when hitting Escape

### ISSUE THREAD

**Issue by mpsijm** [autoNumeric/autoNumeric#782]
Given that `isCancellable: false` and `selectOnFocus: false`:

### Current behavior
When I hit Escape, the input value is selected.

### Expected behavior
When I hit Escape, the input value should not be selected.

### Steps to reproduce the problem
1. Use autoNumeric version `v4.10.4`
2. In both Firefox 123 and Chrome 122
3. On the Arch Linux operating system
4. Then instantiate the autoNumeric object with the following options:
```js
new AutoNumeric(domElement, { isCancellable: false, selectOnFocus: false });
```
5. ~~Then my computer exploded, for real.~~ If I type some numbers and hit Escape, the input value is selected.

Reproduction example: https://jsfiddle.net/Lxh8aned/

### Code exploration
I found [this code](https://github.com/autoNumeric/autoNumeric/blob/next/src/AutoNumeric.js#L6620-L6622):
```js
            // ...and lastly we update the caret selection, even if the option `isCancellable` is false
            this.select();
            //TODO Add an option to select either the integer or decimal part with `Esc`
```
I would expect that the update to the caret selection would at least follow the options that already exist, e.g. `selectOnFocus` and `selectNumberOnly`.

**Comment by AlexandreBonneau** [comment]
I fail to see how `selectOnFocus` is related to the Escape key allowing to select when hitting the Escape key.
Also, as per the documentation, `isCancellable` also allow the Esc key to select everything:
<img width="758" alt="2024-03-06_103612_Screenshot" src="https://github.com/autoNumeric/autoNumeric/assets/7248877/ccb6ad64-acea-485e-bfb6-19496b3cf9c2">

So those options and behaviors are working as expected, and will not change.

However there could be a use case where hitting the Esc key only cancel the changes (or not), and do not select anything.

**Comment by mpsijm** [comment]
Yes, it feels like those two options could be separated, thanks for the elaboration :slightly_smiling_face:
